### ISSUE TITLE
Allowing subscribers to search the archives of a newsletter (https://secure.helpscout.net/conversation/2306561578/1129/)

### ISSUE THREAD

**Issue by jmduke** [buttondown/roadmap#2950]
## TL;DR 

Add search to the web archives. Self-explanatory and overdue, I've just punted on it because it's a bit of a hairy feature and I didn't want to back myself into building a CMS.

## Breakdown

- [ ] Backend support. I think we can start off by being very naive with this: just hit Django's ORM directly, do FTS like we do in `providers/online/email.py` and wrap that in a view.
- [ ] Frontend support. I trust your judgment on design stuff, but can supply a mock if need be; I imagine basically the same implementation everyone else does (there's a search button in the topnav that opens a modal, you type in some stuff and the modal is populated with results.)
- [ ] Add a flag to the `Newsletter` model, defaulting to True, that allows authors to opt out of search if they prefer.
- [ ] Add docs!
- [ ] Add changelog entry!

## Notes

- The classic web archive template can be considered out of scope for this — only `v2` needs to implement the frontend!
- Probably the biggest complicating factor is that the `v2` archives don't really have an obvious entry point for this to glom onto — the JS they use are either the `SubscribeForm` (a standalone Vue component + endpoint) or raw JS (see `createDialogs`.) If it's trivial to implement this in the latter fashion; that's great. If we need  to wrap another little Vue component like we do with `SubscribeForm`, I also think that's perfectly fine.

## Open questions

TBD.
